plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -686,6 +770,25 @@ def create_vm(module, proxmox, vmid, newid, node, name, memory, cpu, cores, sock
             if p in kwargs:
                 del kwargs[p]
 
+    # Clean up the cloud-init variables
+
+    if 'ipconfig' in kwargs:
+        # Need to 'break down' the ipconfig dictionary into a series of the following
+        # ipconfig0: ip=192.168.1.2/24,gw=192.168.1.1
+        # ipconfig1: ip=192.168.2.2/24
+        for int_number in kwargs['ipconfig']:
```

Comment:
It seems the following code is not fully correct because according your comment the `gw` is optional but the code expects the gw to be set.

This would be my patch:

```python
for int_number, ipconfig_dict in kwargs['ipconfig'].enumarate():
    int_label = str('ipconfig%d' % int_number)

    res = list()
    for k, v in ipconfig_dict.items():
        res.append('%s=%s' % (k, v))

    kwargs[int_label] = ','.join(res)

```

Reply:
You're right, that's part of the copy/pasted code snippets from the old repo :wink: I've updated it accordingly.